src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -593,6 +620,22 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
     if (health())
         return;
 
+    clearSql (*ledgerDb_,
+        "DELETE FROM Validations WHERE LedgerHash IN "
+        "( "
+        "SELECT v.LedgerHash "
+        "FROM Validations v "
+        "LEFT OUTER JOIN Ledgers l "
+        "ON v.LedgerHash = l.LedgerHash "
+        "WHERE l.LedgerHash is NULL "
+        "LIMIT %u "
+        ") "
+        ";"
+        );
```

Comment:
Here's a commit that shows with the edits I had in mind for `clearSql` and `clearCaches`: https://github.com/seelabs/rippled/commit/80750e573d4baa01c451b05d1cbe113aa3a790d4

I'll note the sql is harder to read with the `std::to_string`, so no problem if you want to reject that and keep `boost::format`.